supertokens_python/ingredients/emaildelivery/service/smtp.py

```diff
@@ -46,68 +47,111 @@ def __init__(
         self.auth = auth
 
 
-class GetContentResult(TypedDict):
-    body: str
-    subject: str
-    toEmail: str
+class GetContentResult:
+    def __init__(self, body: str, subject: str, to_email: str) -> None:
+        self.body = body
+        self.subject = subject
+        self.to_email = to_email
 
 
 class TypeInputSendRawEmailFrom:
-    name: str
-    email: str
-
     def __init__(self, name: str, email: str) -> None:
         self.name = name
         self.email = email
 
 
-class TypeInputSendRawEmail(GetContentResult):
-    user_context: Dict[str, Any]
-    config_from: TypeInputSendRawEmailFrom
-
-
 class TypeGetContentInput(Generic[_T]):
     user_context: Dict[str, Any]
 
 
 class ServiceInterface(ABC, Generic[_T]):
     @abstractmethod
-    def send_raw_email(self, input: TypeInputSendRawEmail):
+    async def send_raw_email(self,
+                             get_content_result: GetContentResult,
+                             config_from: TypeInputSendRawEmailFrom,
+                             user_context: Dict[str, Any]
+                             ) -> None:
         pass
 
     @abstractmethod
-    def get_content(self, input: TypeGetContentInput[_T]):
+    def get_content(self, input: _T, user_context: Dict[str, Any]) -> GetContentResult:
         pass
 
 
-class TypeInput(Generic[_T]):
-    smtpSettings: SMTPServiceConfig
+class EmailDeliverySMTPConfig(Generic[_T]):
+    def __init__(self,
+                 smtpSettings: SMTPServiceConfig,
+                 override: Union[Callable[[ServiceInterface[_T]], ServiceInterface[_T]], None] = None
+                 ) -> None:
+        self.smtpSettings = smtpSettings
+        self.override = override
 
-    @abstractmethod
-    def override(self, oi: ServiceInterface[_T]) -> ServiceInterface[_T]:
-        pass
 
+class Transporter:
+    def __init__(self, smtpSettings: SMTPServiceConfig) -> None:
+        self.smtpSettings = smtpSettings
 
-class Transporter():
-    pass
+    def send_email(self, config_from: TypeInputSendRawEmailFrom, get_content_result: GetContentResult, user_context: Dict[str, Any]) -> None:
+        print(config_from)
+        try:
+            smtp = smtplib.SMTP(self.smtpSettings.host, self.smtpSettings.port)
+            smtp.starttls()
+            if self.smtpSettings.auth:
+                smtp.login(self.smtpSettings.auth.user, self.smtpSettings.auth.password)
 
-
-def createTransport(_) -> Transporter:
-    return Transporter()
+            smtp.sendmail(config_from.email, get_content_result.to_email, get_content_result.body)
+            smtp.quit()
+        except Exception as _:
+            pass
 
 
 TypeGetDefaultEmailServiceImpl = Callable[[Transporter, TypeInputSendRawEmailFrom], ServiceInterface[_T]]
 
+class SMTPEmailDeliveryImplementation(EmailDeliveryInterface[_T]):
```

Comment:
remove this if not used

Reply:
Done. Thanks! 

How do you identify unused code so easily?

Reply:
Thanks!

Reply:
Haha, I missed removing this particular function from node from the commit you're referring. So I thought you would also have this function in your code 😆